packages/workspace/src/core/affected-project-graph/locators/implicit-json-changes.ts

```diff
@@ -17,11 +21,20 @@ export function getImplicitlyTouchedProjectsByJsonChanges(
 
   for (const f of touchedFiles) {
     if (f.file.endsWith('.json') && implicitDependencies[f.file]) {
-      const changes: JsonValueDiff[] = f.getChanges();
+      const changes = f.getChanges();
       for (const c of changes) {
-        const projects =
-          getTouchedProjects(c.path, implicitDependencies[f.file]) || [];
-        projects.forEach(p => touched.push(p));
+        if (isJsonChange(c)) {
+          const projects =
+            getTouchedProjects(c.path, implicitDependencies[f.file]) || [];
+          projects.forEach(p => touched.push(p));
+        } else {
+          walkJsonTree(implicitDependencies[f.file], [], (p, value) => {
```

Comment:
what are you trying to do here?

I think this should be extracted into a function otherwise the if and else clauses operate on different levels of abstraction. One talks about touched projects, the other one recursively walks the tree

Reply:
This recursively finds projects affected by a JSON file.

```
implicitDependencies: {
  'package.json': {
    dependencies: {
      '@angular/core': ['proj1'],
    },
    devDependencies: ['proj2']
  }
}
```

Should touch `['proj1', 'proj2']` when touched.

Reply:
Cool. Could you extract it into a helper function saying something like: `collectProjectsAffectedByAllChangesToPackageJson`